Make sure that exit code is propagated from Child to parent process

atuljangra opened the pull request:
Summary: Refractor error_handler.py

Test Plan:
In the previous diff, I added a unit test which showcases the failed case. With this diff, we can see that the override works as expected.

Also added few additional tests for test coverage

Reviewed By: wilson100hong

Differential Revision: D37677402

d4l3k commented on torch/distributed/elastic/multiprocessing/errors/error_handler.py:
```diff
@@ -81,6 +81,25 @@ def record_exception(self, e: BaseException) -> None:
             with open(file, "w") as fp:
                 json.dump(data, fp)
 
+    def override_error_code_in_rootcause_data(
```
can you add a docstring to this?

atuljangra committed:
Make sure that exit code is propagated from Child to parent process (#81408)

Summary:
Refractor error_handler.py

Pull Request resolved: https://github.com/pytorch/pytorch/pull/81408

Test Plan:
In the previous diff, I added a unit test which showcases the failed case. With this diff, we can see that the override works as expected.

Also added few additional tests for test coverage

Reviewed By: wilson100hong

Differential Revision: D37677402

Pulled By: atuljangra

fbshipit-source-id: 3d59d87e14518ae84e9cd4737d118d9cfd84fafe
